### PR TITLE
Upgrade pipenv dependencies

### PR DESCRIPTION
**Pull request by berroar**
### What is the context of this PR?
Upgrades dependencies in order to fix security vulnerabilities.

### How to review
Check nothing is broken by this change. Check that the submissions confirmation consumer is able to be deployed as part of our staging pipeline.

### Checklist
\*[ ] Tests updated

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -179,16 +179,24 @@ def send_email(request: Request) -> Tuple[str, int]:
...
+                **log_context,
+                notify_error=notify_error,
+                status_code=status_code,
+            )
+
+            return message, error.response.status_code
+
+        log_error(message, **log_context)
+
+        return message, None
```
Comment (MebinAbraham) [inline main.py]: How come this was needed?.

- Have you tried this in a deployed env?. It is probably expecting a status code.
- If it is valid, it needs a new test.

File: Pipfile [hunk 2]
```diff
@@ -4,7 +4,7 @@ verify_ssl = true
 name = "pypi"
 
 [packages]
-flask = "*"
+flask = "<3.0.0"
```
Comment (MebinAbraham) [inline Pipfile]: Are you sure this doesn't work `>3`? According to their set-up, they only require `<4`. https://github.com/GoogleCloudPlatform/functions-framework-python/blob/main/setup.py#L53

As per https://cloud.google.com/functions/docs/writing/specifying-dependencies-python
We can define the function framework version. We should move it to the main dependency section as Google recommends installing it as an explicit dependency. For some reason, Google uses framework version 3 when we don't explicitly set one, which is over 2 years old!.

> The [Functions Framework](https://cloud.google.com/functions/docs/functions-framework) is a required dependency for all functions. Although Cloud Functions installs it on your behalf when the function is created, we recommend that you include it as an explicit dependency for clarity.

Comment (berroar) [reply Pipfile]: Updated now 👍

File: main.py (hunk 1 above)
Comment (berroar) [reply main.py]: Reverted as discussed and will raise another card to address the mypy issues and the broken test on main 👍

File: Pipfile (hunk 2 above)
Comment (berroar) [reply Pipfile]: Also removed wrapt 👍

**Review by MebinAbraham — APPROVED** [review]

**Comment by berroar** [comment]
[Dependabot Process PRs](https://trello.com/c/wIOC8rDo/6075-dependabot-process-prs)

**Review by liamtoozer — APPROVED** [review]
Tested in my dev env and works great 👍

**Pull request merged**